vllm/v1/worker/gpu_model_runner.py

```diff
@@ -527,6 +536,12 @@ def _prepare_inputs(
         np.add(self.input_batch.num_computed_tokens_cpu[req_indices],
                arange,
                out=positions_np)
+        # ======== XPU  start =========
+        seq_lens = (self.input_batch.num_computed_tokens_cpu[:num_reqs] +
+                    num_scheduled_tokens)
+        self.seq_start_loc_np[0] = 0
+        np.cumsum(seq_lens, out=self.seq_start_loc_np[1:num_reqs + 1])
+        # ======== XPU end =========
```

Comment:
Is this the only needed hard code for XPU?

Reply:
ipex attn (chunked prefill) use flashattn v2, need this parameters.